### PR TITLE
fix bug on issue view when not login

### LINKED ISSUE #1539: Panic in `routers/repo.ViewIssue()` when not logged in
This URL serves a blank page if you are not authenticated:
https://try.gitea.io/gitea/gitea/issues/6

\cc @bkcsoft @lunny @appleboy @tboerger (someone who can check the logs?)

### PR DESCRIPTION
**Pull request by lunny**
will fix #1539

### CODE REVIEW

File: routers/repo/issue.go [hunk 1]
```diff
@@ -465,16 +465,26 @@ func ViewIssue(ctx *context.Context) {
...
+		}
+		if !exists {
+			iw = &models.IssueWatch{
+				UserID:     ctx.User.ID,
+				IssueID:    issue.ID,
+				IsWatching: models.IsWatching(ctx.User.ID, ctx.Repo.Repository.ID),
+			}
+		}
+	} else {
 		iw = &models.IssueWatch{
```
Comment (sapk) [inline routers/repo/issue.go]: I prefer that iw would stay at `nil` and in template https://github.com/go-gitea/gitea/blob/master/templates/repo/issue/view_content/sidebar.tmpl#L104 the form is only display if logged or/and IssueWatch is set.

Comment (lunny) [reply routers/repo/issue.go]: done.

**Comment by sapk** [comment]
LGTM

**Comment by bkcsoft** [comment]
LGTM

**Comment by bkcsoft** [comment]
@lunny Builds fail though 😢

**Comment by sapk** [comment]
@lunny ~~You need to test the creation of a issue before and next try to view this issue.~~

File: integrations/issue_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+	"code.gitea.io/gitea/models"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNoLoginViewIssue(t *testing.T) {
+	assert.NoError(t, models.LoadFixtures())
+
+	req, err := http.NewRequest("GET", "/user1/repo1/issues/1", nil)
```
Comment (sapk) [inline integrations/issue_test.go]: repo1 is owned by user2 (https://github.com/go-gitea/gitea/blob/master/models/fixtures/repository.yml) -> so /user2/repo1/issues/

Ref : 
- https://github.com/go-gitea/gitea/blob/master/models/fixtures/user.yml
- https://github.com/go-gitea/gitea/blob/master/models/fixtures/issue.yml

Comment (lunny) [reply integrations/issue_test.go]: I'm confusing there are two fixtures. `https://github.com/go-gitea/gitea/blob/master/models/fixtures` and `integrations/gitea-integration/fixtures`,

Comment (sapk) [reply integrations/issue_test.go]: In your test you seem to load `"code.gitea.io/gitea/models".LoadFixtures()` don't know for the other fixture.

**Comment by lunny** [comment]
waiting #1627

**Comment by sapk** [comment]
Good idea of removing duplicate fixture.

**Review by appleboy — APPROVED** [review]

**Comment by lunny** [comment]
let L-G-T-M work

**Pull request merged**